cudax/cmake/cudaxBuildCompilerTargets.cmake

```diff
@@ -1,113 +1,51 @@
-#
 # This file defines the `cudax_build_compiler_targets()` function, which
 # creates the following interface targets:
 #
 # cudax.compiler_interface
-# - Interface target providing compiler-specific options needed to build
-#   cudax's tests, examples, etc.
-
-include("${cudax_SOURCE_DIR}/cmake/AppendOptionIfAvailable.cmake")
+# - Interface target linked into all targets in the CudaX developer build.
+#   This should not be directly used; it is only used to construct the
+#   per-dialect targets below.
+#
+# cudax.compiler_interface_cppXX
+# - Interface targets providing dialect-specific compiler flags. These should
+#   be linked into the developer build targets, as they include both
+#   cudax.compiler_interface and cccl.compiler_interface_cppXX.
 
 function(cudax_build_compiler_targets)
-  set(cxx_compile_options)
   set(cuda_compile_options)
+  set(cxx_compile_options)
+  set(cxx_compile_definitions)
 
   if ("MSVC" STREQUAL "${CMAKE_CXX_COMPILER_ID}")
-    # sccache cannot handle the -Fd option generating pdb files
-    set(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT Embedded)
-
-    list(APPEND cuda_compile_options "--use-local-env")
-
-    append_option_if_available("/W4" cxx_compile_options)
-
-    append_option_if_available("/WX" cxx_compile_options)
-
-    # Suppress overly-pedantic/unavoidable warnings brought in with /W4:
-    # C4324: structure was padded due to alignment specifier
-    append_option_if_available("/wd4324" cxx_compile_options)
-    # C4505: unreferenced local function has been removed
-    # The CUDA `host_runtime.h` header emits this for
-    # `__cudaUnregisterBinaryUtil`.
-    append_option_if_available("/wd4505" cxx_compile_options)
-    # C4706: assignment within conditional expression
-    # MSVC doesn't provide an opt-out for this warning when the assignment is
-    # intentional. Clang will warn for these, but suppresses the warning when
-    # double-parentheses are used around the assignment. We'll let Clang catch
-    # unintentional assignments and suppress all such warnings on MSVC.
-    append_option_if_available("/wd4706" cxx_compile_options)
     # C4848: support for attribute 'msvc::no_unique_address' in C++17 and earlier is a vendor extension
     append_option_if_available("/wd4848" cxx_compile_options)
```

Comment:
We should disable that warning and suppress locally in code

Reply:
Agreed, I see that you've opened #2358 to track this 👍